src/memory/title_sequence_manager.rs

```diff
@@ -0,0 +1,177 @@
+use crate::memory::MemoryManager;
+use crate::state::StateContext;
+use memory::memory_manager::unity::*;
+use memory::process::{Error, Process};
+
+pub struct TitleSequenceManager {
+    pub name: String,
+    pub manager: UnityMemoryManager,
+    pub data: TitleSequenceManagerData,
+}
+
+impl Default for TitleSequenceManager {
+    fn default() -> Self {
+        let manager = Self {
+            name: "TitleSequenceManager".to_string(),
+            data: TitleSequenceManagerData::default(),
+            manager: UnityMemoryManager {
+                ..UnityMemoryManager::default()
+            },
+        };
+        println!("{} Loaded", manager.name);
+        manager
+    }
+}
+
+impl MemoryManager for TitleSequenceManager {
+    fn update_manager(&mut self, ctx: &StateContext) {
+        if let Some(process) = &ctx.process {
+            if let Some(module) = &ctx.module {
+                if let Some(image) = &ctx.image {
+                    self.manager.update(process, module, image, &self.name);
+                }
+            }
+        }
+    }
+
+    fn update_memory(&mut self, ctx: &StateContext) {
+        if self.manager.instance.is_some() {
+            self.data.update(ctx, &mut self.manager)
+        }
+    }
+}
+
+#[derive(Default, Debug)]
+pub struct TitleSequenceManagerData {
+    pub relics: Vec<Relic>,
+    pub title_menu: TitleMenu,
+}
+
+impl TitleSequenceManagerData {
+    pub fn update(&mut self, ctx: &StateContext, manager: &mut UnityMemoryManager) {
+        self.update_title_menu(ctx, manager);
+        self.update_relics(ctx, manager);
+    }
+
+    pub fn update_title_menu(&mut self, ctx: &StateContext, manager: &mut UnityMemoryManager) {
+        if let Some(class) = manager.class {
+            if let Some(process) = &ctx.process {
+                if let Some(module) = &ctx.module {
+                    if let Some(singleton) = manager.singleton {
+                        if class
```

Comment:
I feel there is a cleaner way to do this, but might not be possible due to how the actual memory layout is set up. I'm thinking something like a block (or function) that fetches the fields, and a match statement of some kind that actually sets the `self.title_menu.selected`. Maybe not.